src/main/java/io/neonbee/endpoint/metrics/MetricsEndpoint.java

```diff
@@ -2,19 +2,39 @@
 
 import static io.neonbee.endpoint.Endpoint.createRouter;
 
+import io.micrometer.core.instrument.composite.CompositeMeterRegistry;
+import io.micrometer.prometheus.PrometheusConfig;
+import io.neonbee.NeonBee;
 import io.neonbee.config.EndpointConfig;
 import io.neonbee.endpoint.Endpoint;
 import io.vertx.core.Vertx;
 import io.vertx.core.json.JsonObject;
 import io.vertx.ext.web.Router;
-import io.vertx.micrometer.PrometheusScrapingHandler;
 
 public class MetricsEndpoint implements Endpoint {
+
     /**
      * The default path the metrics endpoint is exposed by NeonBee.
      */
     public static final String DEFAULT_BASE_PATH = "/metrics/";
 
+    /**
+     * Add {@link NeonBeePrometheusMeterRegistry} to the {@link CompositeMeterRegistry}.
+     *
+     * Note that this Method is <b>static synchronized</b> to ensure that the registry is only added once.
+     *
+     * @param vertx the {@link Vertx} instance
+     */
+    @SuppressWarnings("PMD.AvoidSynchronizedAtMethodLevel")
+    private static synchronized void addRegistry(Vertx vertx) {
+        CompositeMeterRegistry compositeMeterRegistry = NeonBee.get(vertx).getConfig().getCompositeMeterRegistry();
+        boolean isNotRegisterd = compositeMeterRegistry.getRegistries().stream()
+                .noneMatch(NeonBeePrometheusMeterRegistry.class::isInstance);
+        if (isNotRegisterd) {
+            compositeMeterRegistry.add(new NeonBeePrometheusMeterRegistry(PrometheusConfig.DEFAULT));
+        }
+    }
+
```

Comment:
remove, the `NeonBeePrometheusMeterRegistry` should be set as default in the `NeonBeeConfig` (instead of initializing the `microMeterRegistries` with a `private List<String> registries = List.of();` initialize it with a `List.of("io.neonbee.endpoint.metric.NeonBeePrometheusMeterRegistry")`. This way it is clear (from a config perspective) which registies get added, it removes the requirement to (with some dubious instance variable) pass the composite metrics registry around via the `getConfig()` method and allows to not add the `NeonBeePrometheusMeterRegistry` if it is unwanted by the user. If there are no registries set in the config, the metrics endpoint should not get deployed in the `NeonBee` class (this way it can be disabled).

The downside of this is: How do you pass configuration from the config into the registry? I think later we should not only have `microMeterRegistries` be a `List<String>`, but add a new `MicroMeterRegistryConfig` object and make it a `List<MicroMeterRegistryConfig>`, that also allows to pass a JSON of options towards the registry implementation.

Reply:
This is kind of the exact opposite of what I had discussed with @pk-work directly. The underlying discussion was this one here https://github.com/SAP/neonbee/pull/94#discussion_r778621669.